tests/unit/subscriptions/test-class-wc-payments-subscription-change-payment-method.php

```diff
@@ -46,6 +46,7 @@ public function test_update_subscription_change_payment_button() {
 		// The update_subscription_change_payment_button function attempts to call WC_Subscription->get_checkout_payment_url(). To avoid errors mock that function's return.
 		$mock_subscription->checkout_payment_url = 'example.com';
 		$mock_subscription->update_meta_data( WC_Payments_Subscription_Service::SUBSCRIPTION_ID_META_KEY, 'sub_test123' );
+		$mock_subscription->set_payment_method( WC_Payment_Gateway_WCPay::GATEWAY_ID );
```

Comment:
Because I changed `WC_Payments_Subscription_Change_Payment_Method_Handler::does_subscription_need_payment_updated()` to use `WC_Payments_Subscription_Service::is_wcpay_subscription( $subscription )` I had to make sure I set the subscription's payment method to WC Payments. 


Previously the logic just checked if the stripe billing`sub_` ID meta was set, whereas `is_wcpay_subscription()` also checks the the subscription's payment method. 

To me this tightens the logic in a way that makes more sense. We shouldn't be overriding these change payment URLs if the subscription's payment method is no longer wcpay.